lmdeploy/tokenizer.py

```diff
@@ -220,6 +221,32 @@ def __init__(self, model_dir: str):
         self.max_indexes_num = 5
         self.token2id = {}
 
+    def _check_transformers_version(self, model_dir: str):
+        import transformers
+        from packaging import version
+        from transformers.models.auto.configuration_auto import \
+            CONFIG_MAPPING_NAMES
+
+        from lmdeploy.utils import get_hf_config_content
+
+        logger = get_logger('lmdeploy')
+
+        current_transformers_version = version.parse(transformers.__version__)
+        cfg = get_hf_config_content(model_dir, trust_remote_code=True)
+        required_transformers_version = version.parse(
+            cfg['transformers_version'])
+        if current_transformers_version < required_transformers_version:
+            logger.warning(
+                f'The current version of `transformers` is transformers=={current_transformers_version}, '  # noqa: E501
+                f'which is lower than the required version transformers=={required_transformers_version}. '  # noqa: E501
+                'Please upgrade to the required version.')
+
+        model_type = cfg['model_type']
+        if model_type not in CONFIG_MAPPING_NAMES.keys():
```

Comment:
This check is not necessary since models with remote code is not in CONFIG_MAPPING_NAMES

Reply:
ok